decayAmplitude/ampIntegralMatrixMetadata.cc

```diff
@@ -228,48 +222,12 @@ bool rpwa::ampIntegralMatrixMetadata::setAllZeroHash() {
 }
 
 
-bool rpwa::ampIntegralMatrixMetadata::addEventMetadata(const rpwa::eventMetadata& evtMeta, size_t eventMin, size_t eventMax) {
-	if (eventMin >= eventMax) {
-		printErr << "given event number bounds are not ordered correctly." << endl;
+bool rpwa::ampIntegralMatrixMetadata::addEventMetadata(const rpwa::eventMetadata& evtMeta) {
+	if (std::find(_evtMetas.begin(), _evtMetas.end(), evtMeta) != _evtMetas.end()) {
+		printWarn << "event metadata object has already been added." << endl;
```

Comment:
I agree, but it does not make much sense to add the same metadata twice (IMHO), it is already there, is it not? To really keep track of what is in the integral file, the metadata have to be save together with the otfBin, but this is beyond the scope of this work. This issue (and several others in the integral metadata) should be revisited when the on-the-fly integrator is decorated with a script.